### PR TITLE
 fix(store): [SSR] - Action dispatch does not work properly

### PR DESCRIPTION
**Pull request by Gorniv**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```
## What is the current behavior?
https://github.com/ngxs/store/issues/696
https://github.com/ngxs/store/issues/465
https://github.com/angular/universal/issues/1061

## What is the new behavior?
run dispatch in ngZone for server side

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by squelix** [comment]
You did better than me, I remove my PR :)

File: integration/app/app.browser.module.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// angular
```
Comment (splincode) [inline integration/app/app.browser.module.ts]: remove comments please

File: integration/app/app.browser.module.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { NgModule } from '@angular/core';
+import { BrowserTransferStateModule } from '@angular/platform-browser';
+// components
```
Comment (splincode) [inline integration/app/app.browser.module.ts]: remove comments please

File: integration/app/app.server.module.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// angular
```
Comment (splincode) [inline integration/app/app.server.module.ts]: delete comments

File: integration/app/app.server.module.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import { NgModule } from '@angular/core';
+import { ServerModule, ServerTransferStateModule } from '@angular/platform-server';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+// libs
```
Comment (splincode) [inline integration/app/app.server.module.ts]: delete comments

File: integration/app/app.server.module.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { ModuleMapLoaderModule } from '@nguniversal/module-map-ngfactory-loader';
+// components
```
Comment (splincode) [inline integration/app/app.server.module.ts]: delete comments

File: integration/test.ssr.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { exit } from 'shelljs';
+const request = require('./request-service');
+
+setTimeout(async () => {
```
Comment (splincode) [inline integration/test.ssr.ts]: could you redo this code to use mocha?
https://www.npmjs.com/package/ts-mocha

```ts
describe('NGXS + SSR`, () => {
  it('Should be exist "NgxsOnInit todo"', () => {
    expect(body.include('NgxsOnInit todo')).toBe(true);
  });
  // ...
});
```

Comment (Gorniv) [reply integration/test.ssr.ts]: No

File: integration/tests-ssr/todo.mocha.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { expect } from 'chai';
+
+const request = require('./request-service');
+let body: string;
+
+describe('NGXS + SSR', () => {
+  // console.log(body);
```
Comment (splincode) [inline integration/tests-ssr/todo.mocha.ts]: Comment?

File: integration/tests-ssr/todo.mocha.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  it('Should be exist "NgxsOnInit todo"', async () => {
+    body = await request('http://localhost:4000/list');
+    return expect(body.includes('NgxsOnInit todo')).to.equal(true);
+  });
```
Comment (splincode) [inline integration/tests-ssr/todo.mocha.ts]: make a line break here

**Comment by splincode** [comment]
@markwhitfeld I'll check everything again, and if something goes wrong, I will make a revert branch!

**Pull request merged**